src/InsaLan/TournamentBundle/Controller/UserController.php

```diff
@@ -416,6 +428,59 @@ public function leaveTeamAction($teamId) {
     }
 
     /**
+     * Allow a captain to ban another player from a team
+     * Only possible if the player didn't pay anything yet OR if this is a free tournament
+     * @Route("/user/ban/team/{teamId}/{playerId}")
+     */
+    public function banPlayerAction($teamId, $playerId) {
+        $em =$this->getDoctrine()->getManager();
+        $team = $em
+            ->getRepository('InsaLanTournamentBundle:Team')
+            ->findOneById($teamId);
+
+        // does the team exist ?
+        if($team === null)
+            return $this->redirect($this->generateUrl('insalan_tournament_user_index'));
+
+        // get current logged user corresponding player
+        $usr = $this->get('security.context')->getToken()->getUser();
+        $captain = $em
+            ->getRepository('InsaLanTournamentBundle:Player')
+            ->findOneByUserAndPendingTournament($usr, $team->getTournament());
+
+        // is he really the captain ? (also check for null)
+        if($team->getCaptain() !== $captain)
+            return $this->redirect($this->generateUrl('insalan_tournament_user_index'));
+
+        $playerToBan = $em
+            ->getRepository('InsaLanTournamentBundle:Player')
+            ->findOneById($playerId);
+
+        // does this player exist ?
+        if($playerToBan === null)
+            return $this->redirect($this->generateUrl('insalan_tournament_user_index'));
+
+        // not possible if the player payed something ! (paid tournament + payement done)
+        if(!$team->getTournament()->isFree() && $player->getPaymentDone())
+            return $this->redirect($this->generateUrl('insalan_tournament_user_index'));
+
+        // captain cannot ban himself !
+        if($playerToBan !== $captain) {
+
+            $playerToBan->leaveTeam($team);
+            $team->removePlayer($playerToBan);
+
+            $em->persist($team);
+
+            $em->remove($playerToBan);
```

Comment:
Need verification, but if you don't remove the player, he/shell will not have to reenter his username and other information. 
It was more critical when other information was League of Legend username verification, which was quite long (players have to create a rune page with a given name, wait 3 minutes, and validate it on insalan.fr to cheick they really own the summoner name they gave)